### PR TITLE
"Added the blockdev command"

### PR DESCRIPTION
**Pull request by bw-space**
I have added the blockdev command with the following options:
--getbsz device
--report [device]
This commit references: #1701

### CODE REVIEW

**Comment by anton-bondarev** [comment]
Thanks
Can you prepare PR according to the rules adopted by Embox ( [https://github.com/embox/embox/wiki/Contributing%3A-Git-cheatsheet](https://github.com/embox/embox/wiki/Contributing%3A-Git-cheatsheet ) )

File: src/cmds/blockdev.c [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+int main(int argc, char **argv) {
+    struct block_dev **bdevs, *bdev;
+
+    if (argc < 2) {
+        print_usage();
+        return 0;
+    }
+
+	if (!strcmp(argv[1], "--getbsz")) {
```
Comment (anton-bondarev) [inline src/cmds/blockdev.c]: may be it better use getopt_long()?

Comment (bw-space) [reply src/cmds/blockdev.c]: yes, I will change it to it.

File: src/cmds/blockdev.c [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+	if (!bdev) {
+		printf("Block device \"%s\" not found.\n", dev_name);
+		return 0;
+	}
+
+	return bdev;
+}
+
+int main(int argc, char **argv) {
+    struct block_dev **bdevs, *bdev;
```
Comment (anton-bondarev) [inline src/cmds/blockdev.c]: Please use tabs in start of line as described in Embox-Code-Style ([https://github.com/embox/embox/wiki/Code-Style](https://github.com/embox/embox/wiki/Code-Style))

**Comment by bw-space** [comment]
> Thanks
> Can you prepare PR according to the rules adopted by Embox ( https://github.com/embox/embox/wiki/Contributing%3A-Git-cheatsheet )

Do you want me to squash all the commits into one?

File: src/cmds/blockdev.c [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+						printf("%d\n", bdev->block_size);
+					}
+				}
+				break;
+			case 2:
+				printf("------------------------------------------------------------\n");
+				printf("| %-15s | %-15s | %-20s |\n", "Device Name", "BSZ", "Size");
+				printf("------------------------------------------------------------\n");
+				bdevs = get_bdev_tab();
+				assert(bdevs);
```
Comment (anton-bondarev) [inline src/cmds/blockdev.c]: Please, don't use assert in command. Use 'if' instead

File: src/cmds/blockdev.c [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+					bdev = chk_dev_avail(argv[2]);
+					if (bdev) {
+						printf("| %-15s | %-15d | %-20lld |\n", bdev->name, bdev->block_size, bdev->size);
+						printf("------------------------------------------------------------\n");
+					}
+				}
+				else {
+					bdev = *(bdevs);
+					while(bdev) {
+						printf("| %-15s | %-15d | %-20lld |\n", bdev->name, bdev->block_size, bdev->size);
```
Comment (anton-bondarev) [inline src/cmds/blockdev.c]: two similar lines 71 and 78 create function for them

File: src/cmds/blockdev.c [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+						bdev = *(++bdevs);
```
Comment (anton-bondarev) [inline src/cmds/blockdev.c]: Use code
`	for (i = 0; i < MAX_BDEV_QUANTITY; i++) {
		if (bdevs[i] `
from lsblk
bdevs table can contain NULL elements among not NULL. I mean NULL does not mean that it is the last element

**Comment by anton-bondarev** [comment]
Please rebase and squash commit as described [https://github.com/embox/embox/wiki/Contributing%3A-Git-cheatsheet](https://github.com/embox/embox/wiki/Contributing%3A-Git-cheatsheet )

**Comment by anton-bondarev** [comment]
@bw-space 
Hi, thanks for your contributions!
But if you want to participate in Embox GSoC2020 you should also send email one of mentor and tell a bit about yourself

**Comment by bw-space** [comment]
Yes, I will be sending you the mail by today.

File: src/cmds/blockdev.c (hunk 3 above)
Comment (bw-space) [reply src/cmds/blockdev.c]: Ok, I will change and update it.

File: src/cmds/blockdev.c (hunk 4 above)
Comment (bw-space) [reply src/cmds/blockdev.c]: Ok,

File: src/cmds/blockdev.c (hunk 5 above)
Comment (bw-space) [reply src/cmds/blockdev.c]: Sure I am changing it.

File: src/cmds/blockdev.c [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+		if (argc >= 3) {
+			bdev = chk_dev_avail(argv[2]);
+			if (bdev) {
+				print_device_info(count,bdev);
+				printf("------------------------------------------------------------------\n");
+			}
+		}
+		else {
+			bdev = *(bdevs);
+			while (bdev) {
```
Comment (anton-bondarev) [inline src/cmds/blockdev.c]: I saw you corrected this while (change to for <MAX_BDEV) but now I see the same mistake.

**Comment by anton-bondarev** [comment]
Now don't rebase and don't squash commit after review start. Just commit ahead your branch.
Now I see the problem which was already discussed ( [https://github.com/embox/embox/pull/1706#discussion_r383704278](https://github.com/embox/embox/pull/1706#discussion_r383704278) ) Please fix it

File: src/cmds/blockdev.c (hunk 6 above)
Comment (bw-space) [reply src/cmds/blockdev.c]: Sorry, I did not understand then, I will correct and send it in some time.

**Comment by bw-space** [comment]
Yes, I am working on it.

**Comment by bw-space** [comment]
I have committed the new changes, please review them once.

File: src/cmds/blockdev.c [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+int main(int argc, char **argv) {
+    struct block_dev **bdevs, *bdev;
+	int count = 0;
+
+    if (argc < 2) {
+        print_usage();
+        return 0;
+    }
+
+	if (!strcmp(argv[1], "--getbsz")) {
```
Comment (anton-bondarev) [inline src/cmds/blockdev.c]: Can you use getopt_long to arguments parsing?

File: src/cmds/blockdev.c [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+		if (bdevs == NULL) {
+			return 0;
+		}
+		if (argc >= 3) {
+			bdev = chk_dev_avail(argv[2]);
+			if (bdev) {
+				print_device_info(count,bdev);
+				printf("------------------------------------------------------------------\n");
+			}
+		}
```
Comment (anton-bondarev) [inline src/cmds/blockdev.c]: Code-style should be
} else {

File: src/cmds/blockdev.c [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+}
+static inline void print_device_info(int count, struct block_dev *bdev) {
+	printf("| %-3d | %-15s | %-15d | %-20lld |\n", count, bdev->name, bdev->block_size, bdev->size);
+}
+
+int main(int argc, char **argv) {
+    struct block_dev **bdevs, *bdev;
+	int count = 0;
+
+    if (argc < 2) {
```
Comment (anton-bondarev) [inline src/cmds/blockdev.c]: check code-style please
Should use only tabs at the start of the line

**Comment by Deryugin** [comment]
@bw-space hey! How is it going?

File: src/cmds/blockdev.c (hunk 8 above)
Comment (bw-space) [reply src/cmds/blockdev.c]: ok got it

File: src/cmds/blockdev.c (hunk 7 above)
Comment (bw-space) [reply src/cmds/blockdev.c]: yes sure.

File: src/cmds/blockdev.c (hunk 9 above)
Comment (bw-space) [reply src/cmds/blockdev.c]: sure, I am doing that.

**Comment by bw-space** [comment]
Actually, we had tests going on, so couldn't find much time, I will surely make the necessary changes and update it soon.

**Comment by anton-bondarev** [comment]
@bw-space
I've created a rebased PR with your changes ##2008 We'll merge it to master soon.  Thanks

**Comment by anton-bondarev** [comment]
Commits merged with #2008

**Pull request closed without merging**